llvm/lib/TableGen/TGParser.cpp

```diff
@@ -3665,6 +3671,42 @@ bool TGParser::ParseDefset() {
   return false;
 }
 
+/// ParseDeftype - Parse a defvar statement.
+///
+///   Deftype ::= DEFTYPE Id '=' Type ';'
+///
+bool TGParser::ParseDeftype() {
+  assert(Lex.getCode() == tgtok::Deftype);
+  Lex.Lex(); // Eat the 'deftype' token
+
+  if (Lex.getCode() != tgtok::Id)
+    return TokError("expected identifier");
+
+  std::string TypeName = Lex.getCurStrVal();
```

Comment:
Can this be `const std::string &` which I believe is what getCurStrVal returns.

Reply:
I can make it `const` but it can't be a reference. `TypeName` will be used in line 3702 and its reference string (which is a field of `Lexer`) has changed when we run to that line.